File: backend/internal/model/plot.go
package model

import "time"

// Plot 菜园地块实体（GIS 展示 + 认养）。
type Plot struct {
	ID          uint      `gorm:"primaryKey" json:"id"`
	Name        string    `gorm:"size:128;not null" json:"name"`
	Code        string    `gorm:"size:32;uniqueIndex;not null" json:"code"`
	Area        float64   `gorm:"not null" json:"area"`
	SoilType    string    `gorm:"size:32;not null" json:"soil_type"`
	Sunlight    string    `gorm:"size:32;not null" json:"sunlight"`
	Latitude    float64   `gorm:"not null" json:"latitude"`
	Longitude   float64   `gorm:"not null" json:"longitude"`
	Status      string    `gorm:"size:32;not null;default:available;index" json:"status"`
	AdopterID   *uint     `gorm:"index" json:"adopter_id"`
	Adopter     *User     `gorm:"foreignKey:AdopterID" json:"adopter"`
	Description string    `gorm:"size:512" json:"description"`
	CreatedAt   time.Time `json:"created_at"`
	UpdatedAt   time.Time `json:"updated_at"`
}

File: frontend/src/api/plot.ts
import { get, post } from '@/utils/request'
import type { UserInfo } from './auth'

export interface Plot {
  id: number
  name: string
  code: string
  area: number
  soil_type: string
  sunlight: string
  latitude: number
  longitude: number
  status: string
  adopter_id: number | null
  adopter: UserInfo | null
  description: string
  created_at: string
}

export interface PlotPayload {
  name: string
  code: string
  area: number
  soil_type: string
  sunlight: string
  latitude: number
  longitude: number
  description?: string
}

export function listPlots(params?: Record<string, any>): Promise<{ list: Plot[]; total: number; page: number; page_size: number }> {
  return get('/plots', { params })
}

export function getPlot(id: number): Promise<Plot> {
  return get(`/plots/${id}`)
}

export function createPlot(payload: PlotPayload): Promise<Plot> {
  return post('/plots', payload)
}

export function adoptPlot(id: number): Promise<Plot> {
  return post(`/plots/${id}/adopt`)
}

export function releasePlot(id: number): Promise<Plot> {
  return post(`/plots/${id}/release`)
}
